src/main/scala/rules/Executor.scala

```diff
@@ -63,74 +64,136 @@ object executor extends ExecutionRules {
+        // If a join point was found, the config argument is enabled,
+        // and the source of both conditional edges was a statement block
+        // (we are at an if-statement), we may join the resulting branches again.
+        case Some(newJoinPoint) if
+          Verifier.config.moreJoins() &&
+          edge1.isInstanceOf[ConditionalEdge[ast.Stmt, ast.Exp]] &&
+          edge2.isInstanceOf[ConditionalEdge[ast.Stmt, ast.Exp]] &&
+          edge1.source.isInstanceOf[StatementBlock[ast.Stmt, ast.Exp]] &&
+          edge2.source.isInstanceOf[StatementBlock[ast.Stmt, ast.Exp]]
+        => {
+          // Here we assume that edge1.condition is the negation of edge2.condition.
+          assert((edge1.condition, edge2.condition) match {
+            case (exp1, ast.Not(exp2)) => exp1 == exp2
+            case (ast.Not(exp1), exp2) => exp1 == exp2
+            case _ => false
+          })
+
+          eval(s, edge1.condition, pvef(edge1.condition), v)((s1, t0, v1) =>
+            joiner.join[scala.Null, scala.Null](s1, v1, resetState = false)((s2, v2, QB) => {
+              brancher.branch(s2, t0, v2)(
+                // Follow only until join point.
+                (s3, v3) => follow(s3, edge1, v3, Some(newJoinPoint))((s, v) => QB(s, null, v)),
+                (s3, v3) => follow(s3, edge2, v3, Some(newJoinPoint))((s, v) => QB(s, null, v))
+              )
+            })(entries => {
+              val s2 = entries match {
+                case Seq(entry) => // One branch is dead
+                  entry.s
+                case Seq(entry1, entry2) => // Both branches are alive
+                  entry1.pathConditionAwareMerge(entry2)
+                case _ =>
+                  sys.error(s"Unexpected join data entries: $entries")
+              }
+              (s2, null)
+            })((s4, _, v4) => {
+              // Continue after merging at join point.
+              exec(s4, newJoinPoint, s4.methodCfg.inEdges(newJoinPoint).head.kind, v4, joinPoint)(Q)
+            })
+          )
+
+        }
+        case _ =>
+          val uidBranchPoint = SymbExLogger.currentLog().insertBranchPoint(edges.length)
+          val res = edges.zipWithIndex.foldLeft(Success(): VerificationResult) {
+            case (fatalResult: FatalResult, _) => fatalResult
+            case (_, (edge, edgeIndex)) => {
+              if (edgeIndex != 0) {
+                SymbExLogger.currentLog().switchToNextBranch(uidBranchPoint)
+              }
+              SymbExLogger.currentLog().markReachable(uidBranchPoint)
+              follow(s, edge, v, joinPoint)(Q)
+            }
           }
-          SymbExLogger.currentLog().markReachable(uidBranchPoint)
-          follow(s, edge, v)(Q)
+          SymbExLogger.currentLog().endBranchPoint(uidBranchPoint)
+          res
       }
-      SymbExLogger.currentLog().endBranchPoint(uidBranchPoint)
-      res
+
+    } else {
+      sys.error("At most two out edges expected.")
```

Comment:
Without joining, Silicon can handle more than two outgoing edges. Can you reasonably restructure the code so that the error is only raised when Option `moreJoins` is enabled?